src/compo/icartt_nc2ioda.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python3
+
+#
+# (C) Copyright 2024 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+import sys
+import argparse
+import netCDF4 as nc
+import numpy as np
+import pandas as pd
+from datetime import datetime, timedelta
+import os
+from pathlib import Path
+
+import xarray as xr
+import math
+from numpy import log as ln
+
+import pyiodaconv.ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from pyiodaconv.orddicts import DefaultOrderedDict
+
+
+# constants
+HPA2PA = 1E2
+ppbv2molmol = 1e-9
+pptv2molmol = 1e-12
+
+# {'iodaName' : ['obsName', 'iodaUnit', 'obsToIodaUnivConv']
+obsvars = {'nitrogendioxideInsitu': ['NO2_ACES', 'mol mol-1', ppbv2molmol],
+           'carbonmonoxideInsitu': ['CO_ppb', 'mol mol-1', ppbv2molmol],
+           'ozoneInsitu': ['O3_CL', 'mol mole-1', ppbv2molmol],
+           'formaldehydeInsitu': ['CH2O_ISAF', 'mol mol-1', pptv2molmol],
+           'airTemperature': ['T', 'K', 1],
+           'windEastward': ['U', 'm s-1', 1],
+           'windNorthward': ['V', 'm s-1', 1]}
+
+
+class icartt(object):
+
+    def __init__(self, filenames):
+
+        self.filenames = filenames
+        self.make_dictionaries()      # Set up variable names for IODA
+        self.DimDict = {}
+        self.read()    # Read data from file
+
+    def read(self):
+
+        # Loop through input filenames
+        first = True
+        for filename in self.filenames:
+            try:
+                dsFlight = xr.open_dataset(filename)
+            except IOError:
+                raise IOError('%s file not found!' % self.filename)
+            except Exception:
+                raise Exception('Unknown error opening %s' % self.filename)
+
+            # Read global attributes
+            self.AttrData['platform'] = dsFlight.attrs['PLATFORM']
+            self.AttrData['description'] = dsFlight.attrs['RA']
+
+            # Read lat lon
+            lats = dsFlight['Latitude'].values
+            lons = dsFlight['Longitude'].values
+
+            lats = lats.astype(np.float32)
+            lons = lons.astype(np.float32)
+
+            # Read time and conver to ioda time format
+            # sec since SDATE
+            stime = dsFlight['iWAS_Start_UTC'].values
+            sdate_str = dsFlight.attrs['SDATE']
+            sdate = pd.to_datetime(sdate_str, format='%Y, %m, %d')
+            epoch = pd.Timestamp('1970-01-01T00:00:00Z').tz_convert(None)
```

Comment:
is pandas only used for this time stuff?  is there overhead bringing in both pandas and xarray, think this can just be done with `datetime` ??
```suggestion
            sdate = datetime.strptime(sdate_str, '%Y, %m, %d')
```